test/test_model_emulation.jl

```diff
@@ -132,4 +132,22 @@ end
     var2 = read_variable(results, ActivePowerVariable, ThermalStandard)
 
     @test var1 == var2
+
+    # Not currently supported.
+    # Deserialize with different optimizer attributes.
+    #optimizer = JuMP.optimizer_with_attributes(
+    #    OSQP.Optimizer,
+    #    "verbose" => true,
+    #    "max_iter" => 50000,
+    #)
+    #@test_logs (:warn, r"Original solver used attribute") match_mode = :any EmulationModel(
+    #    path,
+    #    optimizer,
+    #)
```

Comment:
This requested use case doesn't work. PSI code creates optimizer attributes outside of JuMP, not within it. I don't see a way to list the custom solver attributes that have been set (from JuMP). If you guys know of a way to do that, I could check them. Otherwise, I'm stuck.

Reply:
@daniel-thom you are correct, see this: 

![image](https://user-images.githubusercontent.com/16385323/134092485-c817579e-dbc1-45e6-a1f7-82a84963b3ea.png)

Reply:
My takeaway is that I should delete the code related to optimizer attributes and only log the name. Yours?

Reply:
We can still capture the attributes but only if the optimizer passed is an `optimizer_with_attributes` object. Otherwise we can't. We can do something like: 

if no attributes -> assume that the solver config was the default and otherwise capture the attributes.

Reply:
@daniel-thom we keep a copy of the optimizer in the settings: 

```Julia
struct Settings
    horizon::Base.RefValue{Int}
    time_series_cache_size::Int
    warm_start::Base.RefValue{Bool}
    initial_time::Base.RefValue{Dates.DateTime}
    optimizer::Union{Nothing, MOI.OptimizerWithAttributes}
    direct_mode_optimizer::Bool
    optimizer_log_print::Bool
    system_to_file::Bool
    export_pwl_vars::Bool
    allow_fails::Bool
    ext::Dict{String, Any}
end
```

On capturing the version of the solver we are kind of out of luck since MOI doesn't have a solver version attribute like a name that we can directly use. It shouldn't be too hard to contribute that into the JuMP-dev ecosystem since many wrappers (most of the ones I checked) support a call to return the solver version. 

if a user uses a custom configuration, those parameters will be captured there. We can just do 

```
if !isempty(settings.optimizer.params)
    atttributes = deepcopy(settings.optimizer.params)
else
   attributes = "Solver used with default parameters" 
end
```

Reply:
1. Isn't that a private attribute subject to change? Shouldn't we only use public/exported methods?
2. params is defined as `params::Vector{Pair{AbstractOptimizerAttribute,Any}}`. Is this guaranteed to be serializable with the Serialization library?
3. I didn't realize that we save a copy of the optimizer in Settings. The user can change it in the call to `run!` or `solve!` but the value in Settings is not updated. Is that a bug? Does this mean that we should detect a change in the optimizer in `run!` and then re-serialize the problem?